server/services/stats/side/tc_client.go

```diff
@@ -0,0 +1,111 @@
+package stats_side
+
+import (
+	"errors"
+	"github.com/GSokol/go-teamcity"
+	"github.com/ivankorobkov/di"
+)
+
+func TcClientModule(m *di.Module) {
+	m.AddConstructor(NewTcClient)
+	m.MarkPackageDep(Config{})
+}
+
+type TCBuildInfo struct {
+	BuildStatuses     []string `json:"lastBuilds"`
+	LastChangesAuthor string   `json:"lastChangesAuthor"`
+	LastChangesDate   string   `json:"lastChangesDate"`
+}
+
+type TcClient interface {
+	GetTCBuildInfo() (*TCBuildInfo, error)
+}
+
+type tcClient struct {
+	client teamcity.Client
+	config Config
+}
+
+const (
+	tcBuildNumber = 5
+)
+
+var ErrTcNoChangesYet = errors.New("No changes yet")
+
+func setTcConfigDefaults(config *Config) {
+	if config.TeamCityBuildNumber == 0 {
+		config.TeamCityBuildNumber = tcBuildNumber
+	}
+}
+
+func NewTcClient(config Config) TcClient {
+	var authorizer teamcity.Authorizer
+	if config.TeamCityAddress == TestTeamCityAddress {
+		return NewTestTcClient()
+	}
+
+	if config.TeamCityLogin == "" && config.TeamCityPassword == "" {
```

Comment:
`GuestAuth` is useless for some operations which require full access, which i think will be implemented later